triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Server.scala

```diff
@@ -105,21 +103,25 @@ class Server(
     }
   }
 
-  private def restartTriggers(triggers: Vector[RunningTrigger]): Either[String, Unit] = {
-    import cats.implicits._ // needed for traverse
-    triggers.traverse_(runningTrigger =>
-      for {
-        trigger <- Trigger.fromIdentifier(compiledPackages, runningTrigger.triggerName)
-      } yield startTrigger(trigger, runningTrigger))
+  private def restartTriggers(triggers: Vector[RunningTrigger])(
+      implicit ec: ExecutionContext): Future[Either[String, Unit]] = {
+    import cats.implicits._
+    triggers.toList.traverse(
+      runningTrigger =>
+        Trigger
+          .fromIdentifier(compiledPackages, runningTrigger.triggerName)
+          .map(trigger => (trigger, runningTrigger))) match {
+      case Left(err) => Future.successful(Left(err))
+      case Right(triggers) =>
+        Future.traverse(triggers)(x => startTrigger(x._1, x._2)).map(_ => Right(()))
+    }
   }
 
-  private def triggerRunnerName(triggerInstance: UUID): String =
-    triggerInstance.toString ++ "-monitor"
-
-  private def getRunner(triggerInstance: UUID): Option[ActorRef[TriggerRunner.Message]] =
-    ctx
-      .child(triggerRunnerName(triggerInstance))
-      .asInstanceOf[Option[ActorRef[TriggerRunner.Message]]]
+  private def getRunner(triggerInstance: UUID): Future[Option[ActorRef[TriggerRunner.Message]]] = {
```

Comment:
`ctx.child` and `ctx.spawn` are not thread safe and need to be accessed directly from a behavior. afaict, this was always an issue but newer versions of akka throw a runtime error here.